java/java.completion/src/org/netbeans/modules/java/completion/JavaCompletionTask.java

```diff
@@ -780,6 +813,13 @@ private void insideClass(Env env) throws IOException {
                 controller.toPhase(Phase.ELEMENTS_RESOLVED);
                 env.addToExcludes(controller.getTrees().getElement(path));
                 addTypes(env, EnumSet.of(INTERFACE, ANNOTATION_TYPE), null);
+            } else if (last != null && last.token().text().toString().equals("permits")) {
```

Comment:
should use `TokenUtilities.textEquals()`.
Please reformat each your change. (Select your changes > Source > Format)